### DISCUSSION TITLE
Add account sender override option to multicall, useContractReads, readContracts

### DISCUSSION THREAD

**Discussion by BoilingSoup** [wevm/wagmi#2860] [Ideas]
Ability to override the sender for multicall seems to be missing.
Viem's `multicall` has this option, but not wagmi's `multicall`, `useContractReads`, `readContracts` functions.

https://viem.sh/docs/contract/multicall.html#account-optional

**Comment by vinayprolitus** [comment]
So this feature is missing?
I was trying to override msg.sender in useContractReads, but unable to do so.
I tried the following but nothing worked.
```
const earnedRewardsFunc = useContractReads({
    contracts: [{}]
    overrides: { from: account.address },
  });

const earnedRewardsFunc = useContractReads({
    contracts: [{}]
    overrides: { account: account.address },
  });
  
  const earnedRewardsFunc = useContractReads({
    contracts: [{}]
   account: account.address ,
  });

  const earnedRewardsFunc = useContractReads({
    contracts: [{}]
   from: account.address ,
  });
```

**Reply by BoilingSoup** [reply]
Now that I think about it, it doesn't make a lot of sense to expose the account override.
Because msg.sender will end up being the multicall contract's address. 

I think the preferred way to solve this is to include a batch read function in your smart contract.

**Reply by vinayprolitus** [reply]
But viem multicall has this feature so wagmi should provide as well.

**Reply by tmm** [reply]
The next major version of Wagmi will include all Viem options. Hang tight until then!

**Reply by jxom** [reply]
Multicall shouldn't accept the `account` property for the reasons that @BoilingSoup said. Will remove.

**Reply by vinaykharayat** [reply]
Thats the exact reason I want to override.
What if inside read function I am using msg.sender?

**Reply by BoilingSoup** [reply]
> Thats the exact reason I want to override.
> What if inside read function I am using msg.sender?

Do you know how multicall works? It's an external contract that calls the functions for you; you can't override that. 


You can create your own function for calling multiple reads in your contract.

**Reply by vinaykharayat** [reply]
how does viem do this then?

**Reply by BoilingSoup** [reply]
> how does viem do this then?

I haven't tried but I assume it works similary to providing an account override to `useContractRead`.

msg.sender starts as the account you provide, but when it goes through the multicall contract msg.sender will become the multicall contract.

**Reply by vinaykharayat** [reply]
Yes, thats the issue I am facing. I think I need to loop useContractRead now 😪,
if wagmi uses viem under the hood (or anythinv idk) and viem has this feature. Wagmi should also atleast give what they are giving

**Reply by BoilingSoup** [reply]
I agree it should be consistent, but in both our cases the account override would not help. I'm fine with this being removed from viem.
